src/lib/pages/frontsite/SelfCheckout/SelfCheckout.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import Overridable from 'react-overridable';
+import { Container, Header } from 'semantic-ui-react';
+import { BarcodeScanner } from '@components/BarcodeScanner';
+import { Html5Qrcode, Html5QrcodeSupportedFormats } from 'html5-qrcode';
+import { SelfCheckoutModal } from './SelfCheckoutModal';
+import { invenioConfig } from '@config';
+import _get from 'lodash/get';
+
+class SelfCheckout extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      scanner: null,
```

Comment:
I think that it does not make much sense to put `scanner` in the state. Why not on `this` instead?